api/envoy/config/core/v3/protocol.proto

```diff
@@ -94,6 +94,12 @@ message QuicProtocolOptions {
   // Probes the peer at the configured interval to solicit traffic, i.e. ACK or PATH_RESPONSE, from the peer to push back connection idle timeout.
   // If absent, use the default keepalive behavior of which a client connection sends PINGs every 15s, and a server connection doesn't do anything.
   QuicKeepAliveSettings connection_keepalive = 5;
+
+  // An array of QUIC connection options represented in a serialized string, separated by commas.
+  string connection_options = 6;
```

Comment:
Since this is a proto, I wonder if this should be a repeated string, one per option. That would see a bit more idiomatic than a text-delimited string.

Reply:
Great question! I thought about it too when I wrote this and I believe using a single string might be slightly better because of the jni layer. 
Here's a comparison of the 2 approaches:
**Use repeated strings**: The application calls addConnectionOption() multiple times->The options get stored in a Java array->The Java list gets converted to a jni 2d byte array via a for loop->The 2d byte array gets converted to a vector<string> via another for loop->The vector is iterated to populate the proto field->The protos are read into the actual QuicTagVector via a loop.
**Use a serialized string**: The application is responsible for serializing the copt string->The string gets passed all the way down to the string proto field->The string is deserialized by calling ParseQuicTagVector().

We slightly reduce the readability here so that we save computations on doing conversions.

Reply:
Ah, ok, that makes sense. Thanks for the explanation!